crates/templates/src/run.rs

```diff
@@ -248,6 +249,18 @@ impl Run {
                         Err(anyhow::anyhow!("Spin doesn't know what to do with a 'component' snippet outside an 'add component' operation")),
                 }
             },
+            "variables" => {
+                match &self.options.variant {
+                    TemplateVariantInfo::AddComponent { manifest_path } =>
+                        Ok(RenderOperation::MergeToml(
+                            manifest_path.clone(),
+                            MergeTarget::Application("variables"),
+                            content,
+                        )),
+                    TemplateVariantInfo::NewApplication =>
+                        Err(anyhow::anyhow!("Spin doesn't know what to do with a 'variables' snippet outside an 'add component' operation")),
```

Comment:
This error assumes that a Spin CLI user is familiar enough with the templates system to know what a snippet is. Maybe instead:

```suggestion
                        Err(anyhow::anyhow!("Cannot create new applications from templates with variables")),
```

Reply:
That particular error would be misleading - we can totally create new applications from templates with variables - but yeah I can try to think of something better.  (This is meant to be a can't-happen scenario; it can only happen if the template author ships an always-borked template.)

Reply:
Trying "Invalid template: template wants to merge variables into existing manifest when creating a new manifest"

Reply:
No worries, Kate, and thanks for digging into this - I will return to the original message.